syntax = "proto3";

package tarefas;

service TarefaService {
    rpc ListarTarefas (Empty) returns (TarefaList);
    rpc CriarTarefa (Tarefa) returns (Tarefa);
    rpc CriarTarefaStream (stream Tarefa) returns (TarefaList);
}

message Tarefa {
    string id = 1;
    string titulo = 2;
    string descricao = 3;
    string estado = 4;
    string data_criacao = 5;
    string data_limite = 6;
}

message TarefaList {
    repeated Tarefa tarefas = 1;
}

message Empty {}
